### ISSUE TITLE
Some XML entities not decoded when converting to other formats

### ISSUE THREAD

**Issue by pkmnfrk** [oozcitak/xmlbuilder2#142]
**Describe the bug**
Entities like `&amp;`, `&gt;`, etc remain encoded when converting to eg object or json formats. Curiously, the quote entites (`&quot;` and `&apos;`) seem to translate fine

**To Reproduce**
This is easy to reproduce during an interactive session:
```
> const create = require("xmlbuilder2").create;
undefined
> create("<test>&lt;test&gt;</test>").end({format: "json"});
'{"test":"&lt;test&gt;"}'
> create("<test>pb&amp;j</test>").end({format: "json"})
'{"test":"pb&amp;j"}'
> create("<test>&quot;double quotes&quot;</test>").end({format: "json"})
'{"test":"\\"double quotes\\""}'
> create("<test>&apos;single quotes&apos;</test>").end({format: "json"})
`{"test":"'single quotes'"}`

> create().ele("test").txt("pb&j").end({format: "xml"})
'<?xml version="1.0"?><test>pb&amp;j</test>'
> create(create().ele("test").txt("pb&j").end({format: "xml"})).end({format: "object"})
{ test: 'pb&amp;j' }
```

**Expected behavior**
I expect `&amp;` to translate to `&`, `&gt;` to translate to `>`, etc

**Version:**
 - node.js: 14.17.1
 - xmlbuilder2: 3.0.2

**Comment by Nooxlock** [comment]
Any update on this issue? Trying to find a temporary solution but with little to no success...

**Comment by universalhandle** [comment]
This looks like a duplicate of various issues that have been opened related to ampersand encoding (see #105, #109, #110). I believe this to be fixed with the release of [v3.1.0](https://github.com/oozcitak/xmlbuilder2/releases/tag/v3.1.0). Please reopen if you can still reproduce after upgrading.
